### PR TITLE
E_CLASSROOM-296 [FE] Create a reusable table component

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-296
### Definition of Done
- [x] Create another component for DataTable, so that other pages can reuse this component
- This component should accept parameters for header and row data
- [x] Update quizzes page to apply with the new DataTable

#### Main note
- http://localhost:3003/admin/quizzes?page=1
#### Pages Affected
- 

### Scenarios/ Test Cases
- N/A
#### User Interface
- N/A
#### User Experience 
- N/A
#### Functionality
- N/A
### Screenshots
- N/A

### CODE REVIEW

**Comment by catampongan** [comment]
## Approved PR #120 
- [x] Able to display data and there are no changes on the table

------------------
image
-----------------
- Total Test Time: 0.25 hr
- Total Iterations: 0
- Total Comments: 0

File: src/components/DataTable/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    </Table>
+  );
+};
+
+DataTable.propTypes = {
+  tableHeaderNames: PropTypes.array,
+  renderTableData: PropTypes.func
+};
+
+export default DataTable;
```
Comment (karlolee50) [inline src/components/DataTable/index.js]: put extra space
```suggestion
export default DataTable;

```

File: src/components/DataTable/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+import style from './index.module.scss';
+
```
Comment (karlolee50) [inline src/components/DataTable/index.js:6]: please put a comment to briefly explain how to use this component and what kind of data are to be passed as props

File: src/components/DataTable/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    &:nth-child(4) {
+        width: 151px;
+        text-align: center;
+    }
+
+    &:nth-child(5) {
+        width: 151px;
+        text-align: center;
```
Comment (karlolee50) [inline src/components/DataTable/index.module.scss]: what would happen if there are more than 5 headers? please change approach.

maybe the header width can be styled in the component that calls the DataTable component

File: src/pages/admin/quizzes/QuizList/index.js [hunk 4]
```diff
@@ -92,15 +100,11 @@ const QuizList = () => {
...
-                <FaRegEdit size="20px" />
-              </Button>
+              <FaRegEdit size="20px" color="black"/>
             </Link>
           </td>
           <td id={style.buttonColumn}>
-            <Button className={style.designButton}>
-              <RiDeleteBin2Fill size="30px" color="#db7771" />
-            </Button>
+            <RiDeleteBin2Fill size="30px" color="#db7771" />
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizList/index.js:107]: the edit and column buttons are present in almost all the tables. is is possible that we can move this to the DataTable component?

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 20 minutes
Comment/s: 4

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**